examples/src/world-to-screen.ts

```diff
@@ -0,0 +1,157 @@
+  const renderer = new THREE.WebGLRenderer();
+  const canvas = renderer.domElement;
+  renderer.setPixelRatio(window.devicePixelRatio);
+  renderer.setClearColor('#444');
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(canvas);
+
+  const { htmlElement, paragraph } = createHtmlElements();
+  document.body.appendChild(htmlElement);
+
+  let pickingNeedsUpdate = false;
+  const pickedNodes: Set<number> = new Set();
+  const shading = reveal_threejs.createDefaultShading({
+    color(treeIndex: number) {
+      if (pickedNodes.has(treeIndex)) {
+        return [255, 255, 0, 255];
+      }
+      return undefined;
+    }
+  });
+
+  const scene = new THREE.Scene();
+  const cadNode = new CadNode(cadModel, { shading });
+  const { position, target, near, far } = cadNode.suggestCameraConfig();
+  const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, near, far);
+  const controls = new CameraControls(camera, renderer.domElement);
+
+  controls.setLookAt(position.x, position.y, position.z, target.x, target.y, target.z);
+  controls.update(0.0);
+  camera.updateMatrixWorld();
+  scene.add(cadNode);
+
+  let updateUI = true;
+  const worldPosition: Vector3 = new Vector3(0, 0, 0);
+  createGUI(worldPosition, htmlElement, () => {
+    updateUI = true;
+  });
+  const clock = new THREE.Clock();
+  const render = async () => {
+    const delta = clock.getDelta();
+    const controlsNeedUpdate = controls.update(delta);
+    const sectorsNeedUpdate = await cadNode.update(camera);
+
+    if (controlsNeedUpdate || sectorsNeedUpdate || pickingNeedsUpdate) {
+      renderer.render(scene, camera);
+    }
+    if (controlsNeedUpdate || updateUI) {
+      updateUI = false;
+      const { x, y } = reveal.worldToViewport(canvas, camera, worldPosition);
+      htmlElement.style.top = `${y}px`;
+      htmlElement.style.left = `${x}px`;
+    }
+
+    requestAnimationFrame(render);
+  };
+
+  const pick = (event: MouseEvent) => {
+    if (event.button === MOUSE.RIGHT) {
+      return;
+    }
+    const rect = renderer.domElement.getBoundingClientRect();
+    const coords = {
+      x: ((event.clientX - rect.left) / renderer.domElement.clientWidth) * 2 - 1,
+      y: ((event.clientY - rect.top) / renderer.domElement.clientHeight) * -2 + 1
+    };
+    // Pick in Reveal
+    const revealPickResult = (() => {
+      const intersections = reveal_threejs.intersectCadNodes([cadNode], { renderer, camera, coords });
+      if (intersections.length === 0) {
+        return;
+      }
+
+      // scene.add(createSphere(intersections[0]!.point, 'purple'));
+
+      return intersections[0];
+    })();
+    const treeIndex = revealPickResult!.treeIndex;
+    paragraph.textContent = `treeIndex: ${treeIndex}`;
+    if (!pickedNodes.has(treeIndex)) {
+      pickedNodes.add(treeIndex);
+    } else {
+      pickedNodes.delete(treeIndex);
+    }
+    shading.updateNodes([treeIndex]);
+    pickingNeedsUpdate = true;
+  };
```

Comment:
I would really wish that picking didn't involve this amount of boilerplate, but thats not part of this PR...